homeassistant/components/wmspro/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+from wmspro.webcontrol import WebControlPro
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .config_flow import setup_and_verify
+from .const import DOMAIN, MANUFACTURER
+
+PLATFORMS: list[Platform] = [Platform.COVER]
+
+type WebControlProConfigEntry = ConfigEntry[WebControlPro]  # noqa: F821
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: WebControlProConfigEntry
+) -> bool:
+    """Set up wmspro from a config entry."""
+    hub = await setup_and_verify(hass, entry.data[CONF_HOST])
```

Comment:
This is a shared function, not config flow specific.

Reply:
But we should not share a function with the config flow, they should live independently

Reply:
Okay, since you recommended to inline it anyway below, I can do this in both places.

I thought it would be better to make sure both config and setup follow the same procedure to avoid unaligned behavior (also in the case of future changes / enhancements of this function).